### PR TITLE
Add extension to store oneOf/anyOf/allOf objects as full properties

### PR DESCRIPTION
**Pull request by zippolyte**
oneOf, anyOf and allOf in a composed schema only store the names of the different objects, so it's not possible to do complex things in the templates depending on things like type of object (things like `isArray`, `isEnum`, `isPrimitive`, ...).

This PR converts the objects to properties and stores them in the vendor extensions `x-(oneOf|anyOf|allOf)-as-properties` to be used in templates.

cc @wing328 @jimschubert, would appreciate some feedback on this, thanks 🙂 

I plan on using them in custom templates for now, so I have no template change in this PR, but I feel this could be useful to have here.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.3.x`, `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Review by therve — APPROVED** [review]

**Comment by wing328** [comment]
@zippolyte thanks for the enhancement. I'll review later this week and let you know if I've any questions.

**Comment by zippolyte** [comment]
@wing328 I just merged master hoping it fixes the circle ci build 🙂

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 1]
```diff
@@ -2552,16 +2559,23 @@ public CodegenModel fromModel(String name, Schema schema) {
...
+                        allOfProps.add(interfaceProperty);
                     } else {
                         LOGGER.error("Composed schema has incorrect anyOf, allOf, oneOf defined: {}", composed);
                     }
                 }
             }
 
+            m.vendorExtensions.put("x-oneOf-as-properties", oneOfProps);
+            m.vendorExtensions.put("x-allOf-as-properties", allOfProps);
+            m.vendorExtensions.put("x-anyOf-as-properties", anyOfProps);
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:2577]: I'll rename these to oneOfProps, anyOfProps and allOfProps instead after merging this PR.

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team as the change covers default codegen as well.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:2577]: Done via https://github.com/OpenAPITools/openapi-generator/pull/10231

**Pull request merged**